miraclelinux/meta-debian-extended
# systemd: Update from 241-7~deb10u8 to 241-7~deb10u9

# Purpose of pull request

This version fixes following CVEs.
- CVE-2023-26604

DLA: DLA 3377-1

# Test
1. Build package
2. Run image and test the package on the qemuarm64 environment

## How to test the package
1. Build qemuarm64 image
2. Run qemu image
3. Check systemd

### Build qemuarm64 image
Add the following to local.conf and build qemuarm64 image.
```
DISTRO_FEATURES_append = " systemd"
DISTRO_FEATURES_BACKFILL_CONSIDERED += "sysvinit"
VIRTUAL-RUNTIME_init_manager = "systemd"
VIRTUAL-RUNTIME_initscripts = "systemd-compat-units"
```
```
$ bitbake core-image-minimal
```

### Run qemu image
```
$ runqemu qemuarm64 nographic
```

### Check systemd
Check init program is systemd and SysVinit scripts does not exist.
```
root@qemuarm64:~# ls -la /sbin/init 
root@qemuarm64:~# ps | grep systemd
root@qemuarm64:~# ls -F /etc/
```

## Test result
### Build package
Build succeeded.
```
$ bitbake systemd
Checking Debian Release ...
Sources.xz is not changed.
Force regenerate source code information.
Parsing Debian Sources.xz ...
Parsing recipes: 100% |##############################################################################################################################################| Time: 0:00:28
Parsing of 1362 .bb files complete (0 cached, 1362 parsed). 2384 targets, 80 skipped, 6 masked, 0 errors.
NOTE: Resolving any missing task queue dependencies

Build Configuration:
BB_VERSION           = "1.42.0"
BUILD_SYS            = "x86_64-linux"
NATIVELSBSTRING      = "universal"
TARGET_SYS           = "aarch64-emlinux-linux"
MACHINE              = "qemuarm64"
DISTRO               = "emlinux"
DISTRO_VERSION       = "2.7"
TUNE_FEATURES        = "aarch64 armv8a crc"
TARGET_FPU           = ""
meta                 
meta-yocto-bsp       = "HEAD:d4b57c68b22027c2bedff335dee06af963e4f8a8"
meta-debian          = "HEAD:04469d4b27cb5203de34211f3fd8248a510b6c35"
meta-debian-extended = "update-systemd_241-7+deb10u9:6be13ed77622447e49e2ee456129c9b36faf7846"
meta-emlinux         = "HEAD:c0c24adb1bdb7f4e479976541314e9d9f1bb1f10"
meta-emlinux-private = "HEAD:b290dafea47dfbd8df7ee9b04af38c9b73c87b68"

Initialising tasks: 100% |###########################################################################################################################################| Time: 0:00:01
Sstate summary: Wanted 17 Found 0 Missed 17 Current 409 (0% match, 96% complete)
NOTE: Executing SetScene Tasks
NOTE: Executing RunQueue Tasks
NOTE: Tasks Summary: Attempted 1969 tasks of which 1927 didn't need to be rerun and all succeeded.
```

### Package test
The init program is running systemd.
The SysVinit scripts does not exist.
```
root@qemuarm64:~# ls -la /sbin/init 
lrwxrwxrwx    1 root     root            22 Apr 17 06:44 /sbin/init -> ../lib/systemd/systemd
root@qemuarm64:~# ps | grep systemd
    1 root      9300 S    {systemd} /sbin/init
  112 root     15660 S    /lib/systemd/systemd-journald
  128 systemd- 81488 S    /lib/systemd/systemd-timesyncd
  136 root     13436 S    /lib/systemd/systemd-udevd
  137 systemd-  7860 S    /lib/systemd/systemd-networkd
  141 messageb  4328 S    /usr/bin/dbus-daemon --system --address=systemd: --nofork --nopidfile --systemd-activation --syslog-only
  149 root      7372 S    /lib/systemd/systemd-logind
  153 systemd-  7628 S    /lib/systemd/systemd-resolved
  206 root      3220 S    grep systemd
root@qemuarm64:~# ls -F /etc/
busybox.links.nosuid  group-                issue.net             modprobe.d/           passwd                services              timestamp
busybox.links.suid    gshadow               ld.so.cache           modules-load.d/       profile               shadow                tmpfiles.d/
dbus-1/               gshadow-              ld.so.conf            motd                  protocols             shells                udev/
default/              host.conf             limits                mtab@                 resolv-conf.systemd@  skel/                 udhcpc.d/
depmod.d/             hostname              login.access          nsswitch.conf         resolv.conf@          sysctl.d/             version
fstab                 hosts                 login.defs            os-release@           rpc                   syslog.conf           xattr.conf
group                 issue                 machine-id            pam.d/                securetty             systemd/              xdg/
```
